File: 백준/Gold/1484. 다이어트/다이어트.py
import sys
input = sys.stdin.readline

g = int(input())

answer = []
start = 1
for end in range(1, 100_001):
    cur_g = end**2 - start**2
    while cur_g > g:
        start += 1
        cur_g = end**2 - start**2
        if start > end:
            break
    if cur_g == g:
        answer.append(end)

if not answer:
    print(-1)
else:
    for i in answer:
        print(i)

File: 백준/Gold/1484. 다이어트/다이어트.js
const input = require('fs').readFileSync(process.platform === 'linux' ? '/dev/stdin' : __dirname + '/input.txt').toString().split('\n')

const G = Number(input[0])
const MAX_N = 50_000

let answer = []
let left = 1
let right = 2
while (left < right && right <= MAX_N) {
  const cur_g = right * right - left * left

  if (cur_g < G) right++
  else if (G < cur_g) left ++
  else {
    answer.push(right)
    left++
  }
}

if (!answer.length) console.log(-1)
else console.log(answer.join('\n'))